akka-stream/src/main/scala/akka/stream/javadsl/Source.scala

```diff
@@ -58,7 +58,7 @@ object Source {
    * that mediate the flow of elements downstream and the propagation of
    * back-pressure upstream.
    */
-  def from[O](publisher: Publisher[O]): javadsl.Source[O, Unit] =
+  def from[O](publisher: Publisher[_ >: O]): javadsl.Source[O, Unit] =
```

Comment:
This is wrong: a `Publisher[O]` is most adequately represented as a `Source[O]`. Then if a `Source[O2 >: O]` is required the Java way is to fix that at the use-site of the Source and not here. In any case, if we were to do something here then it would need to be `Publisher[_ <: O]` because otherwise our Source would sometimes emit things outside of its allowed type bound.